.adr/0010-interest-bearing-channel.md

```diff
@@ -0,0 +1,128 @@
+2. security implications
+3. efficiency / gas consumption implications
+4. programmability / tunability of implementation
+5. ergonomics for
+   - channel application developers
+   - SCN wallet developers
+   - network participants
+6. software lock-in, protocol flexibility, composability
+
+... plus the usual meta-considerations such as engineering effort.
+
+# Considered Options
+
+## Fee Buckets as a Protocol Primitive
+
+> (See related [Notion Doc](https://www.notion.so/statechannels/RFC-20-Introduce-a-fee-model-for-intermediaries-by-modifying-reclaim-dce28ac74d764fcf93db7d6b5cf05b3a))
+>
+> The linked document describes a mechanism where a channel _payer_ (Alice) fronts a specified (potential run-time negotiated) fee for each funded virtual channel, which is distributed among route intermediaries at channel closing time. This is enforced at the protocol level via a modification to `reclaim`.
+
+Against the above decision criteria: (on 😀 🙂 😐 🙁 😢
+scale)
+
+1. Fit: 😐 Per channel fees which aren't time-based recreate some of the existing issues from LN, which, as discussed, are still more important in the SCN context.
+2. Security: 🙁 Modification to `reclaim` affects security of the protocol at large - more code in the public API of the adjudicator is greater surface area for bugs, exploits, etc.
+3. Efficiency: 🙂 Probably reasonably strong here - the changes to `reclaim` are only relevant during the sad path, and in this case the accounting is pretty straightforward - checking that outcomes match the prefund commitment for fee distribution.
+4. Flexibility: 🙂 It's relatively easy in this model to adjust fees on the fly and also to adjust based on the number of hops involved in the virtual fund.
+5. Ergonomics:
+   - AppDev: 😀 no effect on channel application development
+   - Wallet: 😐 adds some complexity to the virtual-fund / virtual-defund operations
+   - Users: 🙁 Networking fees originating at the service consumer-side can potentially introduce extra UI hurdles (eg, think of setting gas fees for an L1 transaction). This can be obfuscated from the user by building network fees into the total service fee, but this creates potential for information asymmetries in the marketplace which aren't to the user's benefit.
+6. software-lock: 🙁 Modifications to the adjudicator are the hardest lock-in possible for the network. If this implementation is revealed to be broken, or unfit for purpose, then migration to a separate in-protocol implementation would require a redeployment of the adjudicator, hard shut-down of the entire running SCN, etc. Software-lock could be improved by, eg, implementing claim via upgradable contract, but this introduces tradeoffs in the credible trustlessness of the network.
+
+Meta commentary: sketches toward implementation of this model showed that the engineering, security and auditability effort would be reasonably high for this option.
+
+## Interest Bearing Application Channels
```

Comment:
If you think it is valuable, you could consider discussing the "compound interest" property of solutions 1/2 (i.e. are the fees easily re-deployed to earn more fees themselves).